lib/logstash/filters/elasticsearch/client.rb

```diff
@@ -41,6 +41,17 @@ def search(params)
 
       private
 
+      def setup_hosts(hosts, ssl)
+        hosts.map do |h|
+          if h.start_with?('http:/', 'https:/')
+            h
+          else
+            host, port = h.split(':')
+            { host: host, port: port, scheme: (ssl ? 'https' : 'http') }
```

Comment:
I don't believe this is an issue but can you check that, when ssl is nil, nothing breaks?

Reply:
`config :ssl, :validate => :boolean, :default => false` should give us `true`/`false` values ->
`LogStash::Filters::ElasticsearchClient.new(@logger, @hosts, opts)` (in opts `:ssl => @ssl`) -> 
`ssl = options.fetch(:ssl, false)` here - maybe I misunderstood but this could not break even if `ssl.nil?` (which it isn't atm): the default is already assumed `scheme: 'http'` by Elasticsearch gem ... if `:scheme` isn't set.